### ISSUE TITLE
Crash when deleting an Image in v1

### ISSUE THREAD

**Issue by RichieHindle** [VCVRack/Rack#1299]
Operating system: Window 10 Pro Version 1809.
Version of Rack: v1, 837efa6f0835b2b67f1b9b174caf89c947ab97af (up to date as of May 27th 2019).

When a module uses an `Image`, deleting its `ModuleWidget` crashes Rack because `Image`'s `vg` member is used by the destructor, but `vg` is never initialised.

How to reproduce:

Take a clean Template and add a `PngPanel` like this:

```C++
struct PngPanel : Widget {
    std::shared_ptr<Image> backgroundImage;

    PngPanel() {
        backgroundImage = APP->window->loadImage(asset::plugin(pluginInstance, "res/background.png"));
    }

    void draw(const DrawArgs& args) override {
        NVGpaint paint = nvgImagePattern(args.vg, 0.0, 0.0, box.size.x, box.size.y, 0.0, backgroundImage->handle, 1.0);
        nvgFillPaint(args.vg, paint);
        nvgFill(args.vg);
    }
};

struct MyModuleWidget : ModuleWidget {
    MyModuleWidget(MyModule *module) {
        setModule(module);
        setPanel(APP->window->loadSvg(asset::plugin(pluginInstance, "res/MyModule.svg")));

        PngPanel* pBack = new PngPanel();
        pBack->box.size = box.size;
        addChild(pBack);
        ...
```

Copy the attached background.png to the res directory (or use any old PNG - it doesn't matter).

Start Rack, right-click to see the module browser.  Your module appears.  Click "Fundamental" in the filters list so that the module disappears.  Rack crashes:

```
Thread 1 received signal SIGSEGV, Segmentation fault.
0x000000000042157b in nvgDeleteImage (ctx=0x0, image=20)
    at dep/nanovg/src/nanovg.c:839
839             ctx->params.renderDeleteTexture(ctx->params.userPtr, image);
```

The value of `ctx`, which comes from the `vg` member of the `Image`, is null:
```
(gdb) p ctx
$1 = (NVGcontext *) 0x0
(gdb)
```

...or at least it's uninitialised, so YMMV on that.  For me, it's always null when I follow these steps.

Making `Image::loadFile` store the given `NVGcontext*` in the `vg` member on success makes it work for me, but I wouldn't like to swear that that's the correct fix.

(v0.6 uses the global `gVg` for this, which is why I think this problem is new in v1.)

For the record, here's the full stack trace:
```
(gdb) where
#0  0x000000000042157b in nvgDeleteImage (ctx=0x0, image=20)
    at dep/nanovg/src/nanovg.c:839
#1  0x00000000004538c5 in rack::Image::~Image (this=0x15d86e80,
    __in_chrg=<optimized out>) at src/window.cpp:61
#2  0x0000000000796d0c in __gnu_cxx::new_allocator<rack::Image>::destroy<rack::Image> (this=0x15d86e80, __p=0x15d86e80)
    at C:/msys64/mingw64/include/c++/8.3.0/ext/new_allocator.h:140
#3  0x00000000007aec80 in std::allocator_traits<std::allocator<rack::Image> >::destroy<rack::Image> (__a=..., __p=0x15d86e80)
    at C:/msys64/mingw64/include/c++/8.3.0/bits/alloc_traits.h:487
#4  0x00000000007b4429 in std::_Sp_counted_ptr_inplace<rack::Image, std::allocator<rack::Image>, (__gnu_cxx::_Lock_policy)2>::_M_dispose (this=0x15d86e70)
    at C:/msys64/mingw64/include/c++/8.3.0/bits/shared_ptr_base.h:554
#5  0x0000000008b8880f in std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release (this=0x15d86e70)
    at C:/msys64/mingw64/include/c++/8.3.0/bits/shared_ptr_base.h:155
#6  0x0000000008b887c7 in std::__shared_count<(__gnu_cxx::_Lock_policy)2>::~__shared_count (this=0x15d3a2d8, __in_chrg=<optimized out>)
    at C:/msys64/mingw64/include/c++/8.3.0/bits/shared_ptr_base.h:728
#7  0x0000000008b8868c in std::__shared_ptr<rack::Image, (__gnu_cxx::_Lock_policy)2>::~__shared_ptr (this=0x15d3a2d0, __in_chrg=<optimized out>)
    at C:/msys64/mingw64/include/c++/8.3.0/bits/shared_ptr_base.h:1167
#8  0x0000000008b88248 in std::shared_ptr<rack::Image>::~shared_ptr (
    this=0x15d3a2d0, __in_chrg=<optimized out>)
    at C:/msys64/mingw64/include/c++/8.3.0/bits/shared_ptr.h:103
#9  0x0000000008b87c5a in PngPanel::~PngPanel (this=0x15d3a290,
    __in_chrg=<optimized out>) at src/MyModule.cpp:68
#10 0x0000000008b87c15 in PngPanel::~PngPanel (this=0x15d3a290,
    __in_chrg=<optimized out>) at src/MyModule.cpp:68
#11 0x00000000004a3073 in rack::widget::Widget::clearChildren (
    this=0x15d76b60) at src/widget/Widget.cpp:131
#12 0x00000000004a28fd in rack::widget::Widget::~Widget (this=0x15d76b60,
    __in_chrg=<optimized out>) at src/widget/Widget.cpp:13
#13 0x000000000078cd36 in rack::widget::OpaqueWidget::~OpaqueWidget (
    this=0x15d76b60, __in_chrg=<optimized out>)
    at include/widget/OpaqueWidget.hpp:12
#14 0x0000000000488408 in rack::app::ModuleWidget::~ModuleWidget (
    this=0x15d76b60, __in_chrg=<optimized out>)
    at src/app/ModuleWidget.cpp:248
#15 0x0000000008b84786 in MyModuleWidget::~MyModuleWidget (this=0x15d76b60,
    __in_chrg=<optimized out>) at src/MyModule.cpp:82
#16 0x0000000008b84745 in MyModuleWidget::~MyModuleWidget (this=0x15d76b60,
    __in_chrg=<optimized out>) at src/MyModule.cpp:82
#17 0x00000000004a3073 in rack::widget::Widget::clearChildren (
    this=0x15d4f920) at src/widget/Widget.cpp:131
#18 0x00000000004a28fd in rack::widget::Widget::~Widget (this=0x15d4f920,
    __in_chrg=<optimized out>) at src/widget/Widget.cpp:13
#19 0x000000000078ca46 in rack::widget::ZoomWidget::~ZoomWidget (
    this=0x15d4f920, __in_chrg=<optimized out>)
    at include/widget/ZoomWidget.hpp:10
#20 0x000000000078ca05 in rack::widget::ZoomWidget::~ZoomWidget (
    this=0x15d4f920, __in_chrg=<optimized out>)
    at include/widget/ZoomWidget.hpp:10
#21 0x00000000004a3073 in rack::widget::Widget::clearChildren (
    this=0x15d8e370) at src/widget/Widget.cpp:131
#22 0x00000000004a28fd in rack::widget::Widget::~Widget (this=0x15d8e370,
    __in_chrg=<optimized out>) at src/widget/Widget.cpp:13
#23 0x00000000004a3a4d in rack::widget::FramebufferWidget::~FramebufferWidget
    (this=0x15d8e370, __in_chrg=<optimized out>)
    at src/widget/FramebufferWidget.cpp:13
#24 0x00000000004a3a69 in rack::widget::FramebufferWidget::~FramebufferWidget
    (this=0x15d8e370, __in_chrg=<optimized out>)
    at src/widget/FramebufferWidget.cpp:16
#25 0x000000000078794a in rack::app::ModelBox::deletePreview (this=0x611ae50)
    at src/app/ModuleBrowser.cpp:201
#26 0x0000000000787c2f in rack::app::ModelBox::step (this=0x611ae50)
    at src/app/ModuleBrowser.cpp:207
#27 0x00000000004a31c0 in rack::widget::Widget::step (this=0x6118440)
    at src/widget/Widget.cpp:151
#28 0x00000000004a1031 in rack::ui::SequentialLayout::step (this=0x6118440)
    at src/ui/SequentialLayout.cpp:14
#29 0x00000000004a31c0 in rack::widget::Widget::step (this=0x608c430)
    at src/widget/Widget.cpp:151
#30 0x000000000049db58 in rack::ui::MarginLayout::step (this=0x608c430)
    at src/ui/MarginLayout.cpp:10
#31 0x00000000004a31c0 in rack::widget::Widget::step (this=0x608c070)
    at src/widget/Widget.cpp:151
#32 0x00000000004a31c0 in rack::widget::Widget::step (this=0x6130230)
    at src/widget/Widget.cpp:151
#33 0x000000000049d54d in rack::ui::ScrollWidget::step (this=0x6130230)
    at src/ui/ScrollWidget.cpp:36
#34 0x00000000004a31c0 in rack::widget::Widget::step (this=0x12359e70)
    at src/widget/Widget.cpp:151
#35 0x000000000077f61e in rack::app::ModuleBrowser::step (this=0x12359e70)
    at src/app/ModuleBrowser.cpp:483
#36 0x00000000004a31c0 in rack::widget::Widget::step (this=0x608c7a0)
    at src/widget/Widget.cpp:151
#37 0x0000000000780856 in rack::app::BrowserOverlay::step (this=0x608c7a0)
    at src/app/ModuleBrowser.cpp:132
#38 0x00000000004a31c0 in rack::widget::Widget::step (this=0x6117b10)
    at src/widget/Widget.cpp:151
#39 0x0000000000487909 in rack::app::Scene::step (this=0x6117b10)
    at src/app/Scene.cpp:49
#40 0x000000000045486d in rack::Window::run (this=0x60a4b10)
    at src/window.cpp:364
#41 0x0000000000450f0c in main (argc=2, argv=0x48c18b0) at src/main.cpp:174
```

and here's the PNG that I used:

![background](https://user-images.githubusercontent.com/4448057/58427417-73a32700-8097-11e9-97a5-ce971e9acb0b.png)

**Comment by AndrewBelt** [comment]
Thanks, fixed in 98f3f30

**Comment by RichieHindle** [comment]
Confirmed - thanks for the speedy fix!
